### ISSUE TITLE
use the python interpreter (to set the shebang line) when running pip

### ISSUE THREAD

**Issue by sdementen** [linkedin/shiv#215]
To allow to shiv packages for different versions of python (e.g. 3.8, 3.9, 3.10) from a single version of python (e.g. 3.8) in which shiv is installed, it would be practical to use the python interpreter to use in the shebang line as the python interpreter to use in the "python -m pip install --target ..."

**Comment by lorencarvalho** [comment]
hi @sdementen,
this is an interesting suggestion and one I'm open to, however I would prefer that it be explicit rather than implicit.. perhaps as a flag? `--pip-interpreter` or `--pip-path` ?

**Comment by sdementen** [comment]
Yes indeed. If not from backward incompatibilities, I would have set it as the default as it is the more natural. 
Maybe add a warning for the default case if the version of the two python interpreters (target and runtime) are not the same
